### PR TITLE
settings: Add support for accent-color

### PR DESCRIPTION
**Pull request by A6GibKm**

### CODE REVIEW

**Comment by A6GibKm** [comment]
The biggest change on Color was the addition of the `From<OwnedValue>` otherwise
```
error[E0277]: the trait bound `color::Color: From<OwnedValue>` is not satisfied
   --> src/desktop/settings.rs:178:14
    |
178 |         self.read::<Color>(APPEARANCE_NAMESPACE, ACCENT_COLOR_SCHEME_KEY)
    |              ^^^^ the trait `From<OwnedValue>` is not implemented for `color::Color`
    |
    = note: required for `OwnedValue` to implement `Into<color::Color>`
    = note: required for `color::Color` to implement `TryFrom<OwnedValue>`

For more information about this error, try `rustc --explain E0277`.
warning: `ashpd` (lib) generated 2 warnings
```

File: src/color.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+use crate::zvariant::{self, DeserializeDict, Type};
+use crate::{desktop::screenshot::ColorRequest, Error};
+
+#[derive(DeserializeDict, Clone, Copy, PartialEq, Type)]
+/// The response of a [`ColorRequest`] request.
```
Comment (bilelmoussaoui) [inline src/color.rs:5]: Update the comment

File: src/color.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    /// Blue.
+    pub fn blue(&self) -> f64 {
+        self.color[2]
+    }
+
+    /// Creates a new builder-pattern struct instance to construct
+    /// [`Color`].
+    ///
+    /// This method returns an instance of [`ColorRequest`].
+    pub fn request() -> ColorRequest {
```
Comment (bilelmoussaoui) [inline src/color.rs:33]: This no longer make much sense now, maybe we should figure out how to make it agnostic between settings and color picker?

File: src/desktop/settings.rs [hunk 3]
```diff
@@ -172,6 +173,12 @@ impl<'a> Settings<'a> {
         }
     }
 
+    /// Retrieves the system's preferred accent color
+    pub async fn accent_color(&self) -> Result<Color, Error> {
```
Comment (bilelmoussaoui) [inline src/desktop/settings.rs:177]: Please add receive_accent_color_changed as well

File: src/color.rs (hunk 2 above)
Comment (A6GibKm) [reply src/color.rs:33]: We don't need it to be agnostic. Maybe we can just add a `impl Color` with this single method in the color picker's file

Comment (bilelmoussaoui) [reply src/color.rs:33]: I would keep it in the same file and have a new method that returns the accent color?

File: src/color.rs (hunk 1 above)
Comment (A6GibKm) [reply src/color.rs]: Done.

File: src/desktop/screenshot.rs [hunk 4]
```diff
@@ -172,6 +172,16 @@ impl ColorRequest {
     }
 }
 
+impl Color {
+    /// Creates a new builder-pattern struct instance to construct
+    /// [`Color`].
+    ///
+    /// This method returns an instance of [`ColorRequest`].
+    pub fn pick() -> ColorRequest {
```
Comment (bilelmoussaoui) [inline src/desktop/screenshot.rs:180]: Not enough, the tests would fail :p

Comment (A6GibKm) [reply src/desktop/screenshot.rs:180]: Updated everywhere, except the demo

File: src/color.rs [hunk 5]
```diff
@@ -0,0 +1,72 @@
+use crate::{
```
Comment (bilelmoussaoui) [inline src/color.rs:1]: This file should be inside desktop as it is specific to there. Similar to say HandleToken and so on

**Comment by bilelmoussaoui** [comment]
Thank you

**Pull request merged**